### ISSUE TITLE
Might be broken

### ISSUE THREAD

**Issue by junetried** [junetried/cwcat#1]
In what might be the worst timing ever, a few hours before I created this repo, the game updated. Now it doesn't seem to work.

```text
unexpected element found. Expected: Ebml Found: Cluster
```

I was wondering if this means it's no longer a valid webm, but FFMpeg seems to think it's okay and it still reports libavformat as the muxer. It might be a matroska-demuxer bug?

**Comment by junetried** [comment]
It works *sometimes.* I'm not really sure what's going on. Having done some reading on how Matroska works, I'm more confused now than I was before.

I'm working on a Matroska de/muxing library that will hopefully fix this issue. If I can get another example of this happening, I'll save it and open an issue with matroska-demuxer, since this *seems* to be a bug there?

**Comment by junetried** [comment]
I figured it out. This is due to [my incorrect use of Cursors](https://github.com/junetried/cwcat/blob/be10de1b807d6ae736d401a1f7afce7b23fc7633/src/lib.rs#L95). This will *always* fail, because matroska-demuxer does not rewind the Cursor automatically. To be fair, it probably shouldn't. I'm not sure why I thought it did, now that I give it some thought.

I found that it works "sometimes" because I was actually running an old cwcat-bin binary that was built to use byte slices instead of Cursors.
